src/reactor/block_reporter.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module is the implementation of block reporter reactor.
+
+use futures::future::Either;
+use futures::future::IntoFuture;
+use futures::Future;
+use rlp;
+use std::sync::Arc;
+use web3::contract::Contract;
+use web3::contract::Options;
+use web3::transports::Http;
+use web3::types::Address;
+
+use ethereum::types::block::Block;
+use ethereum::Ethereum;
+use reactor::React;
+use web3::contract::Error;
+
+/// This is gas consumed by report block operation. This is calculated by estimate gas
+/// function from web3.
+const REPORT_BLOCK_ESTIMATED_GAS: i32 = 3_000_000;
+
+pub struct BlockReporter {
+    block_store: Arc<Contract<Http>>,
+    from: Address,
+    event_loop: Box<tokio_core::reactor::Handle>,
+    block_chain: Arc<Ethereum>,
+}
+
+impl BlockReporter {
+    /// Creates a new instance of BlockReporter
+    ///
+    /// # Arguments
+    ///
+    /// * `block_store` - Contract instance of block store.
+    /// * `from` - Address which does block reporting.
+    /// * `event_loop` - The reactor's event loop to handle the tasks spawned by this observer.
+    /// * `block_chain` - Block chain object, where block reporting will happen.
+    pub fn new(
+        block_store: Arc<Contract<Http>>,
+        from: Address,
+        event_loop: Box<tokio_core::reactor::Handle>,
+        block_chain: Arc<Ethereum>,
+    ) -> Self {
+        BlockReporter {
+            block_store,
+            from,
+            event_loop,
+            block_chain,
+        }
+    }
+}
+
+impl React for BlockReporter {
+    /// Defines how different reactor will react on block observation.
```

Comment:
It's more specific in this case and the documentation should reflect that in my opinion. What do you think?